src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngKey.Import.cs

```diff
@@ -35,9 +47,70 @@ public static CngKey Import(byte[] keyBlob, CngKeyBlobFormat format, CngProvider
 
             SafeNCryptProviderHandle providerHandle = provider.OpenStorageProvider();
             SafeNCryptKeyHandle keyHandle;
-            ErrorCode errorCode = Interop.NCrypt.NCryptImportKey(providerHandle, IntPtr.Zero, format.Format, IntPtr.Zero, out keyHandle, keyBlob, keyBlob.Length, 0);
-            if (errorCode != ErrorCode.ERROR_SUCCESS)
-                throw errorCode.ToCryptographicException();
+            ErrorCode errorCode;
+            
+            if (curve == null)
+            {
+                errorCode = Interop.NCrypt.NCryptImportKey(providerHandle, IntPtr.Zero, format.Format, IntPtr.Zero, out keyHandle, keyBlob, keyBlob.Length, 0);
+                if (errorCode != ErrorCode.ERROR_SUCCESS)
+                {
+                    Exception e = errorCode.ToCryptographicException();
+                    if (errorCode == ErrorCode.NTE_NOT_SUPPORTED &&
+                        (format == CngKeyBlobFormat.EccFullPrivateBlob || format == CngKeyBlobFormat.EccFullPrivateBlob))
+                    {
+                        // Attempt to import on < Windows 10
+                        throw new PlatformNotSupportedException(SR.Cryptography_InvalidCurveOid, e);
```

Comment:
OK remove the various checks for format and throw cryptoexception:
                if (errorCode != ErrorCode.ERROR_SUCCESS)
                {
                    throw errorCode.ToCryptographicException();
                }